src/python/pants/engine/native.py

```diff
@@ -257,33 +257,58 @@ def get_build_cflags():
 
 def bootstrap_c_source(output_dir, module_name=NATIVE_ENGINE_MODULE):
   """Bootstrap an external CFFI C source file."""
+
+  tempdir = safe_mkdtemp()
```

Comment:
Might consider using `src/python/pants/util/contextutil.py` `def temporary_dir`.

Reply:
+1

Reply:
Done in dependent PR